README.md

```diff
@@ -21,6 +21,32 @@ Assumptions
 
 `grpcat` assume Python 3. Additionally, at version `0.1` `grpcat` assumes that the compiled protobuf files are in the current working directory.
 
+Testing
+----
+The test program `unit_tests.py` tests `grpcat.py` for the following functionality: valid command parsing from program arguments and a successful RPC server connection. 
+
+Specifically whether the program
+
+- reads in the command line arguments as excpected
+- produces the appropriate error messages on too few or too many commands
+- produces the appropriate error on an invalid target service format
+- fails to make a connection to the RPC server
+- succeeds in connecting to the RPC server
+
+**Usage**
+
+Given a protobuf definition, `hello.proto`, compile it to python:
+
+`$ python3 -m  grpc_tools.protoc  -I. --python_out=. --grpc_python_out=. hello.proto`
+
+Create a `get_hello_request.txt` file with the appropriate message field specified in `hello.proto`, for example `name:"Joe"`. 
+
+Run the following way:
+
+`$ cat get_hello_request.txt | unit_tests.py`
```

Comment:
We shouldn't require the user to create the file to run the unit tests. You should instead:

Create a hello.proto:
`package grpcat;

message Hello {
  required string name = 1;
}`

Compile it, and add the proto file and the resulting generated module to this branch. 

Then, in order to call grpcat as a module inside the unit tests, you would use a mock of fileinput.input to set standard in to the string that would be in get_hello_request.txt. See python 3's `unittest` and `mock` modules (and the `@patch` decorator) to see how to do this. This way the user doesn't have to supply the file to the unit test.